### ISSUE TITLE
Windows 2016 Openssh - xml.etree.ElementTree.ParseError: not well-formed (invalid token)

### ISSUE THREAD

**Issue by DanielFarrarCO** [ansible/awx#6990]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
When running Windows native modules on Win2016 server with OpenSSH (key or password auth) (cmd or powershell types) AWX reports an error: "Unexpected failure during module execution...  The error was: xml.etree.ElementTree.ParseError: not well-formed (invalid token)".  When I run the command directly from the docker container awx_task using ansible-playbook, it executes fine.

##### ENVIRONMENT
* AWX version: 9.3.0, 9.1.1 and 9.0.1
* AWX install method: docker on linux
* Ansible version:  2.9.5 and 2.8.5 
* Operating System: Windows
* Web Browser: Firefox

##### STEPS TO REPRODUCE

Running this via ansible-playbook works, but recreating in AWX fails:

ansible-playbook dir.yml -i hosts --extra-vars ansible_connection=ssh --extra-vars ansible_shell_type=powershell

hosts (with openssh v8.1.0.0p1 Win64 installed):
[windows]
HOSTNAME ansible_user=USERNAME ansible_ssh_private_key_file=path/to/key

dir.yml:
- name: dir
  hosts: windows
  gather_facts: true
  tasks:
          - name: dir
            win_command: 'powershell.exe /c dir'
            register: dir

          - debug: var=dir.stdout_lines

##### EXPECTED RESULTS

Happy Ansible Result (this was ran from the AWX container):

bash-4.4# ansible-playbook dir.yml -i hosts --extra-vars ansible_connection=ssh --extra-vars ansible_shell_type=powershell

PLAY [dir] ***********************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************
ok: [HOSTNAME]

TASK [dir] ***********************************************************************************************************************
changed: [HOSTNAME]

TASK [debug] *********************************************************************************************************************
ok: [HOSTNAME] => {
    "dir.stdout_lines": [
        "",
        "",
        "    Directory: C:\\Users\\...
    ]
}

PLAY RECAP ***********************************************************************************************************************
HOSTNAME               : ok=3    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

##### ACTUAL RESULTS

Mad ansible result from the AWX GUI, though note that the host facts are properly gathered before "failed to connect to the host via ssh".:

Identity added: /tmp/awx_12568_cb9g2klh/artifacts/12568/ssh_key_data (/tmp/awx_12568_cb9g2klh/artifacts/12568/ssh_key_data)
ansible-playbook 2.9.5
  config file = /tmp/awx_12568_cb9g2klh/project/ansible.cfg
  configured module search path = ['/tmp/awx_12568_cb9g2klh/project/library']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.6.8 (default, Oct  7 2019, 17:58:22) [GCC 8.2.1 20180905 (Red Hat 8.2.1-3)]
Using /tmp/awx_12568_cb9g2klh/project/ansible.cfg as config file
host_list declined parsing /tmp/awx_12568_cb9g2klh/tmp3zh1ne8u as it did not pass its verify_file() method
Parsed /tmp/awx_12568_cb9g2klh/tmp3zh1ne8u inventory source with script plugin
[

PLAYBOOK: win_test_dir.yml *****************************************************
1 plays in playbooks/infrastructure/win_test_dir.yml
[

PLAY [dir] *********************************************************************

TASK [Gathering Facts] *********************************************************
task path: /tmp/awx_12568_cb9g2klh/project/playbooks/infrastructure/win_test_dir.yml:1
[
<HOSTNAME> ESTABLISH SSH CONNECTION FOR USER: USERNAME
[
<HOSTNAME> SSH: EXEC ssh -o ControlMaster=no -o ControlPersist=3600s -o ForwardAgent=yes -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="USERNAME"' -o ConnectTimeout=30 -o ControlPath=/tmp/awx_12568_cb9g2klh/cp/%h-%p-%r HOSTNAME 'chcp.com 65001 > $null ; PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand <removed>='
[
<HOSTNAME> (0, b'C:\\\\Users\\\\USERNAME\\\\AppData\\\\Local\\\\Temp\\\\ansible-tmp-1589210950.9389865-168028566246935\\r\\n', b'#< CLIXML\\r\\n<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>')
[
Using module file /usr/local/lib/python3.6/site-packages/ansible/modules/windows/setup.ps1
[
<HOSTNAME> PUT /var/lib/awx/.ansible/tmp/ansible-local-7833q3wb0sw0/tmp_6zu9v7h TO C:\\Users\\USERNAME\\AppData\\Local\\Temp\\ansible-tmp-1589210950.9389865-168028566246935\\AnsiballZ_setup.ps1
[
<HOSTNAME> SSH: EXEC sftp -b - -o ControlMaster=no -o ControlPersist=3600s -o ForwardAgent=yes -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="USERNAME"' -o ConnectTimeout=30 -o ControlPath=/tmp/awx_12568_cb9g2klh/cp/%h-%p-%r '[HOSTNAME]'
[
<HOSTNAME> (0, b'sftp> put /var/lib/awx/.ansible/tmp/ansible-local-7833q3wb0sw0/tmp_6zu9v7h /C:/Users/USERNAME/AppData/Local/Temp/ansible-tmp-1589210950.9389865-168028566246935/AnsiballZ_setup.ps1\\n', b'')
[
<HOSTNAME> ESTABLISH SSH CONNECTION FOR USER: USERNAME
[
<HOSTNAME> SSH: EXEC ssh -o ControlMaster=no -o ControlPersist=3600s -o ForwardAgent=yes -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="USERNAME"' -o ConnectTimeout=30 -o ControlPath=/tmp/awx_12568_cb9g2klh/cp/%h-%p-%r HOSTNAME 'chcp.com 65001 > $null ; PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand <removed>='
[
<HOSTNAME> (1, b'{"changed":false,"ansible_facts":{"ansible_ip_addresses":["HOSTNAME","MACADDR"],"ansible_windows_domain_role":"Primary domain controller","ansible_distribution_version":"10.0.14393.0","ansible_distribution_major_version":"10","ansible_netbios_name":"HOSTNAME","ansible_os_name":"Microsoft Windows Server 2016 Datacenter","ansible_system_description":"","ansible_machine_id":"S-1-5-21-1343805315-..."..."ansible_distribution":"Microsoft Windows Server 2016 Datacenter"..."ansible_reboot_pending":null,"ansible_system":"Win32NT","ansible_owner_name":""}}\\r\\n', b'#< CLIXML\\r\\n#< CLIXML\\r\\n<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs><Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>')
[
<HOSTNAME> Failed to connect to the host via ssh: #< CLIXML\r
#< CLIXML\r
<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs><Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>
[
The full traceback is:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/ansible/executor/task_executor.py", line 146, in run
    res = self._execute()
  File "/usr/local/lib/python3.6/site-packages/ansible/executor/task_executor.py", line 645, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/local/lib/python3.6/site-packages/ansible/plugins/action/gather_facts.py", line 79, in run
    res = self._execute_module(module_name=fact_module, module_args=mod_args, task_vars=task_vars, wrap_async=False)
  File "/usr/local/lib/python3.6/site-packages/ansible/plugins/action/__init__.py", line 923, in _execute_module
    res = self._low_level_execute_command(cmd, sudoable=sudoable, in_data=in_data)
  File "/usr/local/lib/python3.6/site-packages/ansible/plugins/action/__init__.py", line 1071, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/usr/local/lib/python3.6/site-packages/ansible/plugins/connection/ssh.py", line 1196, in exec_command
    stderr = _parse_clixml(stderr)
  File "/usr/local/lib/python3.6/site-packages/ansible/plugins/shell/powershell.py", line 47, in _parse_clixml
    clixml = ET.fromstring(data.split(b"\\r\\n", 1)[-1])
  File "/usr/lib64/python3.6/xml/etree/ElementTree.py", line 1314, in XML
    parser.feed(text)
  File "<string>", line None
xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 1, column 1
fatal: [HOSTNAME]: FAILED! => {
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}
[

PLAY RECAP *********************************************************************
HOSTNAME                : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

##### ADDITIONAL INFORMATION

Windows 2012 works fine with AWX and OpenSSH, only getting this on 2016.

**Comment by ryanpetrello** [comment]
@jborean93 you have any thoughts/pointers on this?

**Comment by ryanpetrello** [comment]
@DanielFarrarCO this looks to be something specific to Ansible's PowerShell plugin, not AWX itself.

**Comment by jborean93** [comment]
I'm only able to replicate this problem when changing the code to disable pipeling on Windows. You can see in your output that it's putting the `setup.ps1` file using SFTP which should not happen for a Windows connection as it supports pipelining just fine. It's running the module just fine but when it goes to process the result's stderr it contains a nested CLIXML message which the code is failing to decode.

We should fix this up, tracking the issue with https://github.com/ansible/ansible/issues/69550, but the way forward for you is to figure out why your task isn't being pipelined. Check to see if you've set any env vars that are set to keep remote files, disable anything to do with pipelining, maybe have set pipelining to false in your ansible.cfg.

**Comment by DanielFarrarCO** [comment]
AH!  @jborean93 you're a wizard, we have that set for our Linux boxes and it's being inherited. Odd that it doesn't impact Windows 2008 or 2012, but does 2016.  I will test a fresh repo tomorrow and confirm back.  Thanks!

**Comment by jborean93** [comment]
It's a problem on hosts that have PowerShell v5 (2016 has by default) as it always outputs a progress record to stderr. This process record is output in the CLIXML form and because 2 powershell instances are being run in the non-pipelines case it causes the nested CLIXML string.

**Comment by DanielFarrarCO** [comment]
Found the root cause of my problem, setting "pipelining=true" at any level was not actually enabling pipelining.  I found "keep_remote_files=true" was set, which was disabling pipelining entirely.  Fix was to disable "keep_remote_files" and enable "pipelining".  

Thanks for pointing me in the right direction!  Not sure if a warning message would be useful if Ansible detects PowerShell v5 while "keep_remote_files" is enabled or if "pipelining" is disabled?  Or if it can just ignore those entirely when not applicable?
